javagen/src/main/java/com/azure/autorest/util/MethodUtil.java

```diff
@@ -55,4 +70,149 @@ public static boolean isMethodIncludeRepeatableRequestHeaders(ProxyMethod proxyM
     public static boolean isHttpMethodSupportRepeatableRequestHeaders(HttpMethod httpMethod) {
         return REPEATABILITY_REQUEST_HTTP_METHODS.contains(httpMethod);
     }
+
+    /**
+     * @param requests a list of requests
+     * @return the first request consumes binary type, if no binary request, return the first request in requests
+     *
+     * If the binary request does not have content-type parameter, we will add one for it
+     */
+    public static Request findFirstBinaryRequest(List<Request> requests, Operation operation) {
+        Request selectedRequest = requests.get(0);
+        for (Request request : requests) {
+            if (request.getProtocol().getHttp().getKnownMediaType() != null
+                    && request.getProtocol().getHttp().getKnownMediaType().equals(KnownMediaType.BINARY)) {
+                // add contentType parameter
+                if (haveDifferentContentTypes(requests) && !hasContentTypeParameter(request)) {
+                    Parameter contentTypeParameter = createContentTypeParameter(request, operation);
+                    request.getParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getParameters()), contentTypeParameter);
+                    if (contentTypeParameter.isRequired()) {
+                        request.getSignatureParameters().add(getBinarySchemaBodyOrContentLengthParameterIndex(request.getSignatureParameters()), contentTypeParameter);
+                    }
+                }
+                selectedRequest = request;
+                break;
+            }
+        }
+        return selectedRequest;
+    }
+
+    /**
+     *
+     * @param request the input request
+     * @return true if there is parameter in the request named "contentType", otherwise, return false
+     */
+    private static boolean hasContentTypeParameter(Request request) {
+        for (Parameter parameter : request.getParameters()) {
+            if (parameter.getLanguage().getDefault().getName() != null && parameter.getLanguage().getDefault().getName().equalsIgnoreCase("contentType")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     *
+     * @param requests a list of requests
+     * @return true if the requests have different content types, otherwise return false
+     */
+    private static boolean haveDifferentContentTypes(List<Request> requests) {
+        Set<KnownMediaType> knownMediaTypes = new HashSet<>();
+        for (Request request : requests) {
+            if (request.getProtocol().getHttp().getKnownMediaType() != null) {
+                knownMediaTypes.add(request.getProtocol().getHttp().getKnownMediaType());
+            }
```

Comment:
what is the different between this and `request.getProtocol().getHttp().getMediaTypes()`?

Reply:
`request.getProtocol().getHttp().getMediaTypes()` returns more specific types, like 'image/jpeg', 'application/pdf' etc.
`request.getProtocol().getHttp().getKnownMediaType()` returns common type, like 'binary'.

Reply:
Then in this method, should we use `getMediaTypes()`?